vm/src/stdlib/itertools.rs

```diff
@@ -488,6 +490,196 @@ mod decl {
         }
     }
 
+    struct GroupByState {
+        current_value: Option<PyObjectRef>,
+        current_key: Option<PyObjectRef>,
+        next_group: bool,
+        grouper: Option<Weak<PyObject<PyItertoolsGrouper>>>,
+    }
+
+    impl fmt::Debug for GroupByState {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            f.debug_struct("PyItertoolsGroupBy")
+                .field("current_value", &self.current_value)
+                .field("current_key", &self.current_key)
+                .field("next_group", &self.next_group)
+                .finish()
+        }
+    }
+
+    impl GroupByState {
+        fn is_current(&self, grouper: &PyItertoolsGrouperRef) -> bool {
+            if let Some(ref current_weak_grouper) = self.grouper {
```

Comment:
I personally prefer not to make simple value (`false` here) tailing chain with deep indent but no matter. (actually I was about to try to convince with below code but lost confidence after watching the result)
```rust
self.grouper.map_or(false, |ref weak| weak.upgrade().map_or(false, |current| grouper.is(&current))
```